### ISSUE TITLE
Certain errors show only when F# file is focused in VS 2017

### ISSUE THREAD

**Issue by BentTranberg** [Microsoft/visualfsharp#3229]
Error messages appear almost immediately in Error List pane when I open and focus one particular file, but stays only as long as I keep that file tab focused.

Screenshot of editor and error list:

![image](https://user-images.githubusercontent.com/15068882/27349449-8af1197a-55f7-11e7-8986-52547413642f.png)

The entire solution compiles successfully, and an application that uses the assemblies in question runs just fine. The problem is only that these errors appear in the Error List pane, apparently for no good reason.

- If the file is closed, the errors go away.
- If another file tab is focused, the errors go away.
- If anything else is focused, the errors go away.
- If the text of that file tab is still visible, but the tab is no longer focused, then the errors go away. This case can be tested thoroughly by having multiple files open in multiple editor panes.

The word "focused" is certainly the correct term here.

The errors will come and go almost immediately most of the time, but sometimes there is a few seconds delay - sufficiently long to remind me it's Intellisense at work here.

However, the first time I observed this, the errors would not go away, so I restarted VS to get rid of them.

When doing a build, the compiler reports success all over, in spite of the errors displayed in the Error List pane. One more reminder it's Intellisense.

The assembly uses another assembly, which effectively only contains the following. The reason for this is that I don't want to put my business F# code in an assembly that is forced to use an earlier F# version (4.3.1.0), which is the case when using this type provider.

So the assembly that only has this, uses F# 4.3.1.0.

```
open FSharp.Data.TypeProviders
type Tredata = DbmlFile<"Tredata.dbml">
```

And everything else uses F# 4.4.0.0.

Is this a compiler bug related to type providers in general? Or is the problem with this particular type provider? Or with the F# version(s)? Does it have something to do with the type provider having to be called at design time? I don't think I am making a mistake reporting, because A) it happens in 2017 but not 2015, and B) there shouldn't be errors in the Error List pane when the compiler reports success and the program runs fine, but could there be exceptions when it comes to type providers and Intellisense?

I may previously have seen other issue reports where actually this issue has been reported but drowned in other information. I am searching for them.

Present in VS 2017 Pro 15.2 (26430.13), F# Nightly 15.4.1.17062002
Not present in VS 2015 Pro 14.0.25431.01 Update 3
.NET Framework 4.5.2
Severity: Only a nuisance.
Note to self: my 912 to create a repro for this. Is a repro wanted/needed?

**Comment by dsyme** [comment]
May be related to #3049

**Comment by dsyme** [comment]
@BentTranberg A standalone repro would help enormously to determine the problem and fix it

**Comment by BentTranberg** [comment]
When working on reducing source for a minimal repro for this, I ran into the problem that an actual error during build is not reported in the Error List pane. Only in the Output pane.

![image](https://user-images.githubusercontent.com/15068882/27355306-d49de7de-560a-11e7-9ff2-8781e91cd32a.png)

The error is that in the expression "sqlPakkedata sort", the sqlPakkedata is no longer a function, but just a string. When investigating this issue, you can add the "sort" parameter as shown in the picture, and see if you have the same problem with the error list.

PS: Repro in a few minutes.

**Comment by BentTranberg** [comment]
When working on reducing source for a minimal repro for this, I discovered the following behavior.

When opening the repro, building the solution, and then opening LegacyReportsTypedSql.fs, the following errors appear.

![image](https://user-images.githubusercontent.com/15068882/27355994-328f6636-560d-11e7-85ae-93bf321e87bd.png)

It's the same errors repeating to the end.

This is not the errors I reported about initially. The behavior is otherwise as reported above.

Then I restart VS. Now I get this.

![image](https://user-images.githubusercontent.com/15068882/27356149-bf0eee7e-560d-11e7-9a0a-98bc249c11cc.png)

We're back to my initial report. This is reproducible.
